model-server/src/main/kotlin/org/modelix/model/server/store/IGenericStoreClient.kt

```diff
@@ -17,8 +17,21 @@
 package org.modelix.model.server.store
 
 import org.modelix.model.IGenericKeyListener
+import org.modelix.model.lazy.RepositoryId
 
-typealias IsolatingStore = IGenericStoreClient<ObjectInRepository>
+interface IsolatingStore : IGenericStoreClient<ObjectInRepository> {
+    /**
+     * Default implementation for removing repository objects.
+     * May be overridden by more efficient, store-specific implementations.
```

Comment:
Should include valid exceptions to be raised when removing fails.

Reply:
We currently don't explicitly throw custom exceptions in any of our StoreClients. We should probably add some to wrap store-specific ones (ignite and sql exceptions) but that would require [MODELIX-799](https://issues.modelix.org/issue/MODELIX-799/Create-a-proposal-on-how-to-organize-exceptions). After that we should apply it to this and the already existing operations.

Reply:
Not totally happy to ignore this right now as the other task will grow with every change we make this way. But ok